### DISCUSSION TITLE
Prefer ‘prefers-color-scheme‘ but allow it to be overridden by the user

### DISCUSSION THREAD

**Discussion by SSmale** [tailwindlabs/tailwindcss#2959] [Help] [answered]
Is there a way to use the system preferences for dark mode but then allow the user to override them if they want? 

I think this would be good for devices that don’t have dark mode but the user wants it

**Comment by simonswiss** [comment, marked as answer]
Hey! 👋

A possible implementation of this is presented in the [Tailwind CSS docs website](https://tailwindcss.com/docs/dark-mode):

```js
// On page load or when changing themes, best to add inline in `head` to avoid FOUC
if (localStorage.theme === 'dark' || (!('theme' in localStorage) && window.matchMedia('(prefers-color-scheme: dark)').matches)) {
  document.querySelector('html').classList.add('dark')
} else {
  document.querySelector('html').classList.remove('dark')
}

// Whenever the user explicitly chooses light mode
localStorage.theme = 'light'

// Whenever the user explicitly chooses dark mode
localStorage.theme = 'dark'

// Whenever the user explicitly chooses to respect the OS preference
localStorage.removeItem('theme')
```

Is that useful?

**Comment by SSmale** [comment]
Thank you for pointing that out! I missed that on my skim of the documentation. This is exactly what I was after

Kind regards,

Simon Smale

**Reply by simonswiss** [reply]
Nice, glad I could help 👍
